doc/remailers.rst

```diff
@@ -0,0 +1,125 @@
+To inform subscribers of its key, the remailer injects an Autocrypt-header into
+outgoing messages. The header format is the same as usual, where the ``addr``
+attribute is the address of the list, and ``keydata`` is their Autocrypt key.
+
+In addition to those two attributes, the remailer SHOULD add the
+``_list-recommendation`` attribute, and MAY add ``_list-recipients`` attribute.
+Both of these attributes are exclusive for use by remailers, and MUST NOT be
+used by regular mailbox addresses. Both of these attributes are non-critical
+(starting with an underscore), which will cause an Autocrypt-capable client that
+does not handle these attributes to skip over them, falling back to the usual
+Autocrypt behavior.
+
+For example::
+
+    Autocrypt: addr=post@list.example; _list-recommendation=mandatory; keydata=XXX
+    Autocrypt: addr=post@list.example; _list-recommendation=available; _list-recipients=7/15; keydata=XXX
+
+The ``_list-recommendation`` attribute SHOULD be included in an Autocrypt header
+sent by a remailer. Its value may be either ``available`` or ``mandatory``. If
+the value is set to ``mandatory``, the remailer requires encryption of every
+message, and will reject plaintext messages. If the value is ``available``,
+encryption is offered as an opt-in mechanism and plaintext messages will be
+handled as usual. In either case, encrypted messages will be delivered
+exclusively to subscribers for whom an encryption key is available.
+
+To inform the decision of whether to encrypt or not, the remailer MAY include a
+``_list-recipients`` attribute. This attribute contains the current number of
+subscribers for which an encryption key is available, and the total number of
+subscribers, separated by a slash character ('/'). This attribute is only useful
+if the ``_list-recommendation`` attribute is set to ``available``.
+
+Administrative Considerations
+----
+
+Graceful Up- and Downgrade
+~~~
+
+To allow graceful up- and downgrade of lists with no intervention by the list
+administrator, a remailer software distribution could automatically include
+Autocrypt-headers with outgoing mail depending on the number of
+Autocrypt-enabled subscribers. For example, a remailer software distribution may
+choose to inject Autocrypt headers only if encryption keys are available for a
+majority of subscribers. Note that obviously, end-to-end encryption is
+unsuitable for lists that are publicly archived.
+
+Missed Message Notifications
+~~~
+
+A remailer MAY send notifications to recipients that were unable to receive an
+encrypted message, if their key was unavailable or unusable for some reason. If
+available, a digest mechanism can be used to aggregate this information and
+avoid redundant notifications.
+
+Limiting Plaintext Exposure
+~~~
+
+An honest remailer SHOULD limit exposure to the plaintext data by re-using the
+session key of the original message and re-encrypting it directly, to avoid
+having the actual plaintext in memory at any time.
```

Comment:
This implies a few things that i think are interesting:

 * the message signature will come from the initial sender, *not* from the mailing list
 * the re-mailer cannot tamper with the content of the message (e.g. no adding footers, no "malware" checking)
 * any internally-protected Autocrypt headers (e.g. "memory-hole") might override whatever Autocrypt headers the re-mailer itself injects.

I'm not sure about the logistical consequences of these ideas.

Reply:
@dkg s first point probably also means that clients who are not aware of this may display the message as having a broken signature.

In general i think a signature by the initial sender with a intended-recipients header would provide the best assurance of integrity. It would signal who the message was send by and that it was indeed intended for the list. However the logic in evaluating such a message seems quite complex - in particular with the headers the remailer has to change for delivery.